doc/adrs/086_allow_single_click_tools_to_be_executed_on_edges.adoc

```diff
@@ -0,0 +1,99 @@
+= ADR-086 - Allow single click tools to be executed on edges
+
+== Context
+
+Today a single click tool can only be executed on Nodes. An enhancement could be to allow single click tools to also be executed on Edges as it is the case in Sirius Desktop.
+
+== Decision
+
+Single click tools can also be executed on Edges.
+
+=== Frontend
+
+In `DiagramRepresentation.types.ts`, a new interface named `DiagramElementDescription` will be introduced.
+`NodeDescription` (there is no `EdgeDescription` in this file) will implement it.
+The target descriptions of a `SingleClickOnDiagramElementTool` will be `DiagramElementDescription` instead of `NodeDescription`.
+
+[source,ts]
+----
+export interface DiagramElementDescription {
+  id: string;
+}
+
+export interface NodeDescription extends DiagramElementDescription {
+  id: string;
+}
+
+export interface SingleClickOnDiagramElementTool extends Tool {
+  appliesToDiagramRoot: boolean;
+  selectionDescriptionId: string;
+  targetDescriptions: DiagramElementDescription[];
+}
+----
+
+=== Backend
+
```

Comment:
The new Java interface `IDiagramElementDescription` is not mentioned.

Reply:
It is mentioned below.